truss/templates/server/model_wrapper.py

```diff
@@ -824,7 +824,7 @@ async def __call__(
                     predict_result,
                     span_predict,
                     detached_ctx,
-                    release_and_end=get_defer_fn(),
+                    get_cleanup_fn=get_defer_fn,
```

Comment:
Previously, just by _requesting_ the `defer_fn`, we were [triggering](https://github.com/basetenlabs/truss/blob/47bc00e6037f6ff0952ae5021c0273e9e30f3813/truss/templates/server/model_wrapper.py?plain=1#L100-L102) an unintended side effect of ending spans even if the underlying deferred fn wasn't invoked. The previous code would only fetch this function in the branch where it knew it needed it, but after consolidating to a shared helper where one branch performed no cleanup we broke the test.

I think it's pretty subtle to have side effects just from requesting the cleanup function without actually invoking it, but this change should preserve the behavior.